test/parallel/test-eventtarget-once-twice.js

```diff
@@ -1,14 +1,24 @@
 'use strict';
 const common = require('../common');
-const { once } = require('events');
+const assert = require('node:assert/strict');
+const { once } = require('node:events');
+const { test } = require('node:test');
+const { setImmediate } = require('node:timers/promises');
 
-const et = new EventTarget();
-(async function() {
-  await once(et, 'foo');
-  await once(et, 'foo');
-})().then(common.mustCall());
+test('should resolve `once` twice', (t, done) => {
```

Comment:
This can be simplified as an asynchronous function, right? Calling "done" in the middle might lead to a race condition that generates extraneous asynchronous activity

Reply:
(Now, after I updated the PR to use async (again), albeit slightly simpler)

Reply:
I'm inclined on the async way, it's much cleaner, and there's IMO less of a chance for a race condition